browser/test/Services/AutoClosingPairsTests.ts

```diff
@@ -1,22 +1,97 @@
 import * as assert from "assert"
 
-import * as AutoClosingPairs from "./../../src/Services/AutoClosingPairs"
+import * as types from "vscode-languageserver-types"
+
+import * as Mocks from "./../Mocks"
+
+import {
+    checkOpenCharacter,
+    getWhiteSpacePrefix,
+    IAutoClosingPair,
+} from "./../../src/Services/AutoClosingPairs"
 
 describe("AutoClosingPairs", () => {
     describe("getWhitespacePrefix", () => {
         it("returns empty if the string doesn't have whitespace", () => {
-            const result = AutoClosingPairs.getWhiteSpacePrefix("test")
+            const result = getWhiteSpacePrefix("test")
             assert.strictEqual(result, "")
         })
 
         it("returns tab", () => {
-            const result = AutoClosingPairs.getWhiteSpacePrefix("\ttest")
+            const result = getWhiteSpacePrefix("\ttest")
             assert.strictEqual(result, "\t")
         })
 
         it("returns spaces", () => {
-            const result = AutoClosingPairs.getWhiteSpacePrefix("  test")
+            const result = getWhiteSpacePrefix("  test")
             assert.strictEqual(result, "  ")
         })
     })
+
+    describe("handleOpenCharacter", () => {
+        let mockEditor: Mocks.MockEditor
+        let mockBuffer: Mocks.MockBuffer
+
+        beforeEach(() => {
+            mockEditor = new Mocks.MockEditor()
+            mockBuffer = new Mocks.MockBuffer("typescript", "test.ts", [""])
+            mockEditor.simulateBufferEnter(mockBuffer)
+        })
+
+        it("inserts both characters of the pair", async () => {
+            const pair = { open: "(", close: ")" } as IAutoClosingPair
+
+            mockBuffer.setLinesSync(["("])
+            checkOpenCharacter(
+                (str: string) => {
+                    mockBuffer.setLinesSync([str])
+                },
+                pair,
+                mockEditor as any,
+                false,
+            )
+
+            const [firstLine] = await mockBuffer.getLines(0, 1)
+
+            assert.strictEqual(firstLine, "()")
```

Comment:
I'm glad we have this core functionality under test now - in addition to the tests exercising the new functionality below 👍